### ISSUE TITLE
Custom Auth Guard with nest-keycloak-connect

### ISSUE THREAD

**Issue by suryaumapathy2812** [ferrerojosh/nest-keycloak-connect#167]
I am trying to create a custom Auth Guard that extends the AuthGuard from nest-keycloak-connect to add functionality for inserting a new user into my database when a user registers and sends a request to my API. However, I am facing issues with importing the `KeycloakMultiTenantService` from the `nest-keycloak-connect` package.

Here's my current implementation of the CustomAuthGuard:

```
import { Injectable, ExecutionContext, Logger } from '@nestjs/common';
import { Reflector } from '@nestjs/core';
import KeycloakConnect from 'keycloak-connect';
import { AuthGuard, KeycloakConnectConfig } from 'nest-keycloak-connect';
import { KeycloakMultiTenantService } from 'nest-keycloak-connect/services/keycloak-multitenant.service';
import { PrismaService } from 'src/common/prisma.service';

@Injectable()
export class CustomAuthGuard extends AuthGuard {
  constructor(
    singleTenant: KeycloakConnect.Keycloak,
    keycloakOpts: KeycloakConnectConfig,
    logger: Logger,
    multiTenant: KeycloakMultiTenantService,
    reflector: Reflector,
    private readonly prisma: PrismaService,
  ) {
    super(singleTenant, keycloakOpts, logger, multiTenant, reflector);
  }

  async canActivate(context: ExecutionContext): Promise<boolean> {
    // First, execute the base AuthGuard canActivate method
    console.log('Entering Custom AuthGuard');

    const baseResult = (await super.canActivate(context)) as boolean;
    console.log(baseResult);

    // If base AuthGuard canActivate returns false, return false immediately
    if (!baseResult) {
      return false;
    }

    // Get the request object
    const request = context.switchToHttp().getRequest();

    // Custom check 
    // If the custom check fails, return false
    // If the custom check succeeds, return true
    const user = await this.prisma.users.findFirst({
      where: { username: request.user.preferred_username },
    });

    if (!user) {
      console.log('User does not exit');

     // code to insert user into db
    }

    return true;
  }
}

```

```
[8:34:17 pm] Starting compilation in watch mode...

[8:34:31 pm] Found 0 errors. Watching for file changes.

Error: Cannot find module 'D:\github\side-projectz\tt-nest\node_modules\nest-keycloak-connect\dist\services\keycloak-multitenant.service'
    at createEsmNotFoundErr (node:internal/modules/cjs/loader:1092:15)
    at finalizeEsmResolution (node:internal/modules/cjs/loader:1085:15)
    at resolveExports (node:internal/modules/cjs/loader:569:14)
    at Function.Module._findPath (node:internal/modules/cjs/loader:643:31)
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:1056:27)
    at Function.Module._load (node:internal/modules/cjs/loader:923:27)
    at Module.require (node:internal/modules/cjs/loader:1137:19)
    at require (node:internal/modules/helpers:121:18)
    at Object.<anonymous> (D:\github\side-projectz\tt-nest\src\guard\custom.guard.ts:5:1)
    at Module._compile (node:internal/modules/cjs/loader:1255:14)
```

It seems that the `KeycloakMultiTenantService` is not directly exposed by the `nest-keycloak-connect` package, and I cannot find a way to import it. Can you please provide guidance on how to properly import and use the `KeycloakMultiTenantService` in my custom Auth Guard, or suggest an alternative approach to achieve my desired functionality?

**Comment by jachinte** [comment]
It may be too late for you, but just in case this is useful for somebody else:
You don't need to extend `AuthGuard`. You can just implement `CanActivate`. Nest will apply the guards sequentially.

**Comment by ferrerojosh** [comment]
Fixed in 133411e
